scheduler/src/main/java/com/flipkart/flux/redriver/service/RedriverService.java

```diff
@@ -103,11 +103,26 @@ private void redrive() {
         long now = System.currentTimeMillis();
         do {
             messages = messageService.retrieveOldest(offset, batchSize);
+            ArrayList<Future<?>> messageRedrived = new ArrayList<>();
             messages.stream().filter(e -> e.getScheduledTime() < now).forEach(e -> {
                 try {
-                    redriverRegistry.redriveTask(e.getTaskId());
-                } catch (Exception ex) {}
+                    messageRedrived.add(asyncRedriveService.submit(() -> {
+                        redriverRegistry.redriveTask(e.getTaskId());
+                    }));
+                } catch (Exception ex) {
+                }
             });
+            boolean allCompleted = false;
+            while (!allCompleted) {
+                allCompleted = true;
```

Comment:
I think it's good to have a small sleep here, else this loop will do many iterations unnecessarily during http call duration.